### PR TITLE
Local Agent used for service list reads (#276)

### PR DESCRIPTION
**Pull request by chemicL**
If defined, the local consul agent is used to read list of services.
Otherwise, a random agent is picked, but only from the cache of agents
already syncing registrations.
This change prevents situations, where contacting a random agent in the
cluster fails, because it has no ACL-DC configured.

### CODE REVIEW

File: consul/agents.go [hunk 1]
```diff
@@ -10,20 +10,21 @@ import (
 	log "github.com/Sirupsen/logrus"
 	"github.com/allegro/marathon-consul/metrics"
 	"github.com/allegro/marathon-consul/utils"
-	consulapi "github.com/hashicorp/consul/api"
 )
 
 type Agents interface {
-	GetAgent(agentAddress string) (agent *consulapi.Client, err error)
+	GetAgent(agentAddress string) (agent *Agent, err error)
```
Comment (janisz) [inline consul/agents.go]: :+1:  For making `Agents` interface more consitent but please do it in a separate PR.

**Comment by Kboxoxx** [comment]
> If defined, the local consul agent is used to read list of services. Otherwise, a random agent is picked, but only from the cache of agents already syncing registrations. This change prevents situations, where contacting a random agent in the cluster fails, because it has no ACL-DC configured.
